### DISCUSSION TITLE
如何手动删除一个 Subscription

### DISCUSSION THREAD

**Discussion by StrongManLi** [apache/iotdb#14605] [Q&A] [answered]
我在测试订阅相关的代码时，创建了一个 testTopic，以及一个 Subscription 订阅一个指定的路径。但是在程序（Java）执行的过程中，在没有调用consumer.unsubscribe(TOPIC);时直接终止了程序。然后在iotdb-cli中执行  show subscriptions 时始终有这个未关闭的 subscription，这个如何删除。直接删除 topic是不行的，因为有subscription。
这个未关闭的subscription 导致了我后续执行的相同的 subscription 无法接收到消息。

**Comment by VGalaxies** [comment, marked as answer]
在目前的版本下，需要使用 Java SDK 创建该 subscription 对应的 consumer，然后手动执行 unsubscribe 操作，才能删除对应的 subscription

后续版本会考虑在命令行中提供 drop subscription 命令

**Reply by StrongManLi** [reply]
好的，谢谢
